src/Microsoft.Extensions.SecretManager.Tools/Internal/ProjectIdResolver.cs

```diff
@@ -0,0 +1,129 @@
+
+            var targetFile = GetTargetFile();
+            var outputFile = Path.GetTempFileName();
+            _disposables.Push(() => TryDelete(outputFile));
+
+            var commandOutput = new List<string>();
+            var commandResult = Command.CreateDotNet("msbuild",
+                new[] {
+                    targetFile,
+                    "/nologo",
+                    "/t:_FindUserSecretsProperty",
+                    $"/p:Project={projectFile}",
+                    $"/p:OutputFile={outputFile}",
+                    $"/p:Configuration={configuration}"
+                })
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .OnErrorLine(l => commandOutput.Add(l))
+                .OnOutputLine(l => commandOutput.Add(l))
+                .Execute();
+
+            if (commandResult.ExitCode != 0)
+            {
+                _logger.LogDebug(string.Join(Environment.NewLine, commandOutput));
+                throw new GracefulException(Resources.FormatError_ProjectFailedToLoad(projectFile));
+            }
+
+            var id = File.ReadAllText(outputFile);
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new GracefulException(Resources.FormatError_ProjectMissingId(projectFile));
+            }
+
+            return id.Trim();
+        }
+
+        public void Dispose()
+        {
+            while (_disposables.Count > 0)
+            {
+                _disposables.Pop().Invoke();
+            }
+        }
+
+        private string GetTargetFile()
+        {
+            var assemblyDir = Path.GetDirectoryName(GetType().GetTypeInfo().Assembly.Location);
+
+            // targets should be in one of these locations, depending on test setup and tools installation
+            var searchPaths = new[]
+            {
+                AppContext.BaseDirectory,
+                assemblyDir, // next to assembly
+                Path.Combine(assemblyDir, "../../tools"), // inside the nupkg
+            };
+
+            var foundFile = searchPaths
+                .Select(dir => Path.Combine(dir, TargetsFileName))
+                .Where(File.Exists)
+                .FirstOrDefault();
+
+            if (foundFile != null)
+            {
+                return foundFile;
+            }
+
+            // back up plan - if the targets file was not found any of the expected locations,
+            // create the temporary file from the embedded version
+            var outputPath = Path.GetTempFileName();
+            using (var resource = GetType().GetTypeInfo().Assembly.GetManifestResourceStream(TargetsFileName))
+            using (var stream = new FileStream(outputPath, FileMode.Create))
+            {
+                resource.CopyTo(stream);
+            }
+
+            // cleanup
+            _disposables.Push(() => TryDelete(outputPath));
+
+            return outputPath;
+        }
+
+        private static void TryDelete(string file)
```

Comment:
When you go to a string for the disposables just inline this method in the dispose